### PR TITLE
cache-naming-char-limit:Document cache naming character limits

### PR DESCRIPTION
**Pull request by dfitzmau**

### CODE REVIEW

**Comment by dfitzmau** [comment]
Hello @dvagnero and @jabolina 
I ported content from 14.0 to the 13.0.x branch. Would you be OK to approve the changes?

14.0 PR: https://github.com/infinispan/infinispan/pull/10270

**Comment by jabolina** [comment]
We didn't backport the limit verification for 13. In this version, the limit exists, but an exception is not thrown. If the limit is reached/exceeded, the cache will malfunction. IIRC from the JIRA ticket, the server may crash.

Guess we can document that, and extremely advert against trying to use long names.

**Comment by dfitzmau** [comment]
> We didn't backport the limit verification for 13. In this version, the limit exists, but an exception is not thrown. If the limit is reached/exceeded, the cache will malfunction. IIRC from the JIRA ticket, the server may crash.
> 
> Guess we can document that, and extremely advert against trying to use long names.

Hi @jabolina . Thank you for feedback. I updated the content accordingly. The cache would function normally, but the site of failure exists with the Infinispan Server that sets the character limitation on a cache name?

File: documentation/src/main/asciidoc/topics/ref_infinispan_caches.adoc [hunk 1]
```diff
@@ -6,6 +6,21 @@ You can create declarative cache configuration in XML, JSON, and YAML format.
...
+{brandname} restricts characters to a maximum of `255` for a cache name or a cache template name.
+If you exceed this character limit, the {brandname} server might abruptly stop without issuing an exception message.
+Write succinct cache names and cache template names.
+====
+
+[IMPORTANT]
+====
+A file system might set a limitation for the length of a file name, so ensure that a cache's name does not exceed this limitation.
+If a cache name exceeds a file system's naming limitation, general operations or initialing operations towards that cache might fail. 
+Write succinct file names.
```
Comment (jabolina) [inline documentation/src/main/asciidoc/topics/ref_infinispan_caches.adoc:20]: We can use the same as before here.

```suggestion
Write succinct cache names and cache template names.
```

**Review by jabolina — CHANGES_REQUESTED** [review]
Just a small change.

File: documentation/src/main/asciidoc/topics/ref_infinispan_caches.adoc (hunk 1 above)
Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/ref_infinispan_caches.adoc]: Thanks, @jabolina . Good catch. Updated.

Comment (dfitzmau) [reply documentation/src/main/asciidoc/topics/ref_infinispan_caches.adoc]: You OK to approve the PR?

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**